pkg/dockerfile/dockerfile.go

```diff
@@ -282,28 +285,37 @@ func MakeKanikoStages(opts *config.KanikoOptions, stages []instructions.Stage, m
 	if err := resolveStagesArgs(stages, args); err != nil {
 		return nil, errors.Wrap(err, "resolving args")
 	}
-	var kanikoStages []config.KanikoStage
-	for index, stage := range stages {
+
+	stages = stages[:targetStage+1]
+	kanikoStages := make([]config.KanikoStage, len(stages))
+	for index := len(stages) - 1; index >= 0; index-- {
```

Comment:
We have to revert the evaluation order as during the transformation we remove ONBUILD commands from base stages.